src/Native/Unix/System.Native/pal_time.h

```diff
@@ -6,19 +6,20 @@
 
 #include "pal_compiler.h"
 #include "pal_types.h"
+#include <time.h>
 
-typedef struct UTimBuf
+typedef struct TimeSpec
```

Comment:
There must be a reason they didn't use `time_t` directly in the old struct. I see that it's type is unspecified in the C standard. Does it ever really change? It either does and you need to change back to int64_t, or it doesn't and you can map it directly (eliminate the struct).  Additionally, `long` in C++ is `int` in C#.

Reply:
>  Does it ever really change ?

It may change in future
```
The fact that it is in seconds is stated by the POSIX specification, so, if you're coding for POSIX-compliant environments, you can rely on that.
The C++ standard also states that time_t must be an arithmetic type.
Anyway, the unix timing system (second since the Epoch) is going to overflow in 2038.
 So, it's very likely that, before this date, C++ implementations will switch to other non-int datatypes (either a 64b int or a more complex datatype). 
Anyway, switching to a 64b int would break binary compatibility with previous code (since it requires bigger variables), and everything should be recompiled. 
Using 32b opaque handles would not break binary compatibility, 
you can change the underlying library, and everything still works, but time_t would not a time in seconds anymore, 
it'd be an index for an array of times in seconds. 
For this reason, it's suggested that you use the functions you mentioned to manipulate time_t values, and do not assume anything on time_t.
```
I will revert back to using to using int64

> Additionally, long in C++ is int in C#.

i tried this earlier also and again to0. Converting it to int is leading to failure of all the tests in 
https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs#L45 
i am not able to find out why

Reply:
> Converting it to int is leading to failure 

Sorry, long is int with the VC compiler. I think it depends on the target architecture with Unix. Use int64_t to always get 64 bits.